src/app/users/users-achievements/users-achievements.service.ts

```diff
@@ -1,19 +1,44 @@
 import { Injectable } from '@angular/core';
 import { CouchService } from '../../shared/couchdb.service';
+import { map } from 'rxjs/operators';
+import { ResourcesService } from '../../resources/resources.service';
+import { ActivatedRoute } from '@angular/router';
 
 @Injectable({
   providedIn: 'root'
 })
 export class UsersAchievementsService {
 
   readonly dbName = 'achievements';
+  resourceIds = [];
+  parent = this.route.snapshot.data.parent;
 
   constructor(
-    private couchService: CouchService
-  ) {}
+    private couchService: CouchService,
+    private resourcesService: ResourcesService,
+    private route: ActivatedRoute
+  ) {
+    this.resourcesService.resourcesListener(this.parent).pipe(map((resources: any) => {
+      resources.map((res: any) => { this.resourceIds.push(res._id); });
```

Comment:
The only array function which should have side-effects (modifying objects outside of its scope) is `forEach`, and we should avoid using that when possible.

Instead of writing this with side effects you can write it like this (return is implied for one line arrow functions):
```
this.resourceIds = resources.map((res: any) => res._id);
```